components/style/str.rs

```diff
@@ -134,6 +135,29 @@ pub fn read_exponent<I: Iterator<Item=char>>(mut iter: Peekable<I>) -> Option<i3
     }
 }
 
+struct PeekableTakeWhile<'a, I: Iterator + 'a, F>(&'a mut ::std::iter::Peekable<I>, F, bool);
```

Comment:
Are we using these anywhere now ? If not then we can remove this and below `impl` block for it